Buddy list: Prep work for adding two section layout

Riken-Shah opened the pull request:
<!-- Describe your pull request here.-->
This PR extracts all the unambiguous commits that we can merge from #20072.

Fixes: <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented on static/js/buddy_data.js:
```diff
@@ -287,9 +289,18 @@ function maybe_shrink_list(user_ids, user_filter_text) {
+        ];
+    }
+    // else
+    return recently_active_users;
```
Am I correct that this change will result in our showing, in this corner case, the alphabetically first N users in the sidebar? I worry that this will still look broken, just in a different way, unless we somehow present "N more users".

Riken-Shah replied:
Hey @timabbott, Sorry I didn't get what you are trying to say? What is the corner case?

timabbott commented:
I merged the couple commits ending in 20f36431ec3a03a78cdda03563f2801fc5cdf00b.

I'll note that the "Remove `compose_fade`" commit is not a "prep commit" we can merge early, since it removes a feature.

Riken-Shah commented:
Rebased, and removed compose fade commit!

alya commented:
> Rebased, and removed compose fade commit!

Just to clarify, we absolutely plan to remove the compose fade feature; @timabbott just wanted to make sure that we don't do it prematurely (i.e. before merging the rest of the PR).

Riken-Shah commented:
> Just to clarify, we absolutely plan to remove the compose fade feature; @timabbott just wanted to make sure that we don't do it prematurely (i.e. before merging the rest of the PR).

Yeah, that's why I haven't removed it from main PR #20072.

alya commented:
@timabbott I think this is ready for another review from you. I'd love to keep making progress on this one!

Riken-Shah commented:
I've rebased the PR to remove the merge conflict. 

> Let me know when this is updated to no longer remove any existing functionality -- that's an essential requirement for a mergeable prep commit series.

I've also removed compose fade commit a long time back is there any other change/commit that you think is removing existing functionality?

timabbott commented:
Closing, as this project was finally merged in the last few weeks. Thanks for all the effort on this @Riken-Shah!
